src/components/application_manager/test/commands/mobile/on_hash_change_notification_test.cc

```diff
@@ -0,0 +1,122 @@
+#include "command_impl.h"
+#include "commands/commands_test.h"
+#include "application_manager/mock_message_helper.h"
+#include "application_manager/smart_object_keys.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+
+namespace strings = application_manager::strings;
+
+using application_manager::commands::mobile::OnHashChangeNotification;
+using application_manager::MockMessageHelper;
+using application_manager::commands::CommandImpl;
+using testing::Mock;
+using testing::Return;
+using testing::ReturnRef;
+using testing::_;
+
+class OnHashChangeNotificationTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  OnHashChangeNotificationTest()
+      : message_helper_(*MockMessageHelper::message_helper_mock()) {}
+  void SetUp() OVERRIDE {
+    Mock::VerifyAndClearExpectations(&message_helper_);
+  }
+
+  void TearDown() OVERRIDE {
+    Mock::VerifyAndClearExpectations(&message_helper_);
+  }
+  MockMessageHelper& message_helper_;
+};
+
+TEST_F(OnHashChangeNotificationTest, Run_ValidApp_SUCCESS) {
+  const uint32_t kConnectionKey = 1u;
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey;
+
+  SharedPtr<OnHashChangeNotification> command =
+      CreateCommand<OnHashChangeNotification>(msg);
+
+  std::string return_string;
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  EXPECT_CALL(*mock_app, curHash()).WillOnce(ReturnRef(return_string));
+  EXPECT_CALL(message_helper_, PrintSmartObject(_)).WillOnce(Return(false));
+  EXPECT_CALL(app_mngr_, SendMessageToMobile(msg, _));
+
+  command->Run();
+
+  ASSERT_EQ(application_manager::MessageType::kNotification,
+            (*msg)[strings::params][strings::message_type].asInt());
+  ASSERT_EQ(CommandImpl::mobile_protocol_type_,
+            (*msg)[strings::params][strings::protocol_type].asInt());
+  ASSERT_EQ(CommandImpl::protocol_version_,
+            (*msg)[strings::params][strings::protocol_version].asInt());
+}
+
+TEST_F(OnHashChangeNotificationTest, Run_InvalidApp_SUCCESS) {
+  const uint32_t kConnectionKey = 1u;
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey;
+
+  SharedPtr<OnHashChangeNotification> command =
+      CreateCommand<OnHashChangeNotification>(msg);
+
+  std::string return_string;
+  MockAppPtr mock_app = CreateMockApp();
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(MockAppPtr()));
+  EXPECT_CALL(*mock_app, curHash()).Times(0);
+  EXPECT_CALL(message_helper_, PrintSmartObject(_)).Times(0);
+  EXPECT_CALL(app_mngr_, SendMessageToMobile(msg, _)).Times(0);
+
+  command->Run();
+
+  ASSERT_EQ(application_manager::MessageType::kNotification,
+            (*msg)[strings::params][strings::message_type].asInt());
+}
```

Comment:
As the opposite of previous test where is checked that certain events occurred / happened after command->Run(); (like check value of (*msg)[strings::params][strings::protocol_type]) - here can be tested that they did not happen, for example that some keys do not exist / was not assigned (*msg)[strings::params].keyExists( XYZ ) or have default values etc. So I think more asserts could be added here.